core/src/main/java/org/bitcoinj/core/Utils.java

```diff
@@ -67,12 +67,17 @@
      * @return numBytes byte long array.
      */
     public static byte[] bigIntegerToBytes(BigInteger b, int numBytes) {
-        byte[] bytes = new byte[numBytes];
-        byte[] biBytes = b.toByteArray();
-        int start = (biBytes.length == numBytes + 1) ? 1 : 0;
-        int length = Math.min(biBytes.length, numBytes);
-        System.arraycopy(biBytes, start, bytes, numBytes - length, length);
-        return bytes;        
+        checkArgument(b.signum() > 0, "We assume only positive numbers");
```

Comment:
I've seen that keys are not allowed to be zero or one, but the method is used somewhere else.

I think we should consider 0. If you agree, I'll push in a few